Undo temporary policy that disallows changes to collection.xml

karenc opened the pull request:
The temporary policy was added because of a corner case that prevented
us from making Press only publish changed content.  Since there will be
a fix for this, we can remove the temporary policy.

----

Part 1/3 of #190

pumazi commented on tests/functional/views/test_legacy_publishing.py:
```diff
@@ -304,16 +299,18 @@ def test_publishing_overwrite_module_litezip(
+#            "error": 'modifying a collection is temporarily disallowed'
+#        }
+#    ]
+#    assert resp.json['messages'] == expected_msgs
```
Missed taking this out. TODO

pumazi replied:
I've been reassured out-of-band that this will come back into a functioning state in a followup PR

pumazi commented on tests/functional/views/test_legacy_publishing.py:
```diff
@@ -365,7 +362,7 @@ def test_publishing_overwrite_collection_litezip(
     )
     # FIXME: uncomment.
     # assert resp.status_code == 204
```
Remove the `FIXME` too. TODO

pumazi commented on tests/functional/views/test_legacy_publishing.py:
```diff
@@ -365,7 +362,7 @@ def test_publishing_overwrite_collection_litezip(
     # FIXME: uncomment.
     # assert resp.status_code == 204
-    assert resp.status_code == 400  # no changes to collection.xml
+    assert resp.status_code == 200
```
Hmm... was this previously 204 and it is now 200?

karenc replied:
Yeah.. I'm not sure what this means

karenc commented:
I updated the changelog because 7.3.0 has been released and so I don't need to group the changes into 8.0.0 anymore.

Also changed the commit author to be Bryan.
